src/notify_telegram.c

```diff
@@ -0,0 +1,871 @@
+      return -1;
+    }
+    recipients = tmp;
+    recipients[recipients_len] = strdup(value);
+    if (recipients[recipients_len] == NULL) {
+      ERROR("notify_telegram: strdup failed.");
+      return -1;
+    }
+    recipients_len++;
+  } else if (strcasecmp(key, "BotToken") == 0) {
+    sfree(bot_token);
+    bot_token = strdup(value);
+    if (bot_token == NULL) {
+      ERROR("notify_telegram: strdup failed.");
+      return -1;
+    }
+  } else if (strcasecmp(key, "ProxyURL") == 0) {
+    sfree(proxy_url);
+    proxy_url = strdup(value);
+    if (proxy_url == NULL) {
+      ERROR("notify_telegram: strdup failed.");
+      return -1;
+    }
+  } else if (strcasecmp(key, "DisableGettingUpdates") == 0) {
+    disable_getting_updates = IS_TRUE(value);
+  } else if (strcasecmp(key, "WebhookURL") == 0) {
+    sfree(webhook_url);
+    webhook_url = strdup(value);
+    if (webhook_url == NULL) {
+      ERROR("notify_telegram: strdup failed.");
+      return -1;
+    }
+  } else if (strcasecmp(key, "WebhookHost") == 0) {
+    sfree(webhook_host);
+    webhook_host = strdup(value);
+    if (webhook_host == NULL) {
+      ERROR("notify_telegram: strdup failed.");
+      return -1;
+    }
+  } else if (strcasecmp(key, "WebhookPort") == 0) {
+    sfree(webhook_port);
+    webhook_port = strdup(value);
+    if (webhook_port == NULL) {
+      ERROR("notify_telegram: strdup failed.");
+      return -1;
+    }
+  } else if (strcasecmp(key, "MHDDaemonHost") == 0) {
+#if MHD_VERSION >= 0x00090000
+    sfree(mhd_daemon_host);
+    mhd_daemon_host = strdup(value);
+    if (mhd_daemon_host == NULL) {
+      ERROR("notify_telegram: strdup failed.");
+      return -1;
+    }
+#else
+    ERROR("notify_telegram: Option `MHDDaemonHost' not supported. Please "
+          "upgrade libmicrohttpd to at least 0.9.0");
+    return -1;
+#endif
+  } else if (strcasecmp(key, "MHDDaemonPort") == 0) {
+    char *endptr;
+    errno = 0;
+    int tmp = (int)strtol(value, &endptr, /* base = */ 10);
+    if (errno != 0 || endptr == value || tmp < 1 || tmp > 65535) {
+      ERROR("notify_telegram: converting MHDDaemonPort failed.");
+      return -1;
+    }
+    mhd_daemon_port = (unsigned short)tmp;
+  } else {
+    ERROR("notify_telegram: unknown config key.");
+    return -1;
+  }
+  return 0;
+}
+
+static int notify_telegram_notification(const notification_t *n,
+                                        user_data_t __attribute__((unused)) *
+                                            user_data) {
+  char buf[MAX_BUF_SIZE] = "";
+  char *buf_ptr = buf;
+  int buf_len = sizeof(buf);
+  int status;
+
+  status = snprintf(
+      buf_ptr, buf_len, "<b>Notification:</b>\nseverity = %s",
+      (n->severity == NOTIF_FAILURE)
+          ? "FAILURE"
+          : ((n->severity == NOTIF_WARNING)
+                 ? "WARNING"
+                 : ((n->severity == NOTIF_OKAY) ? "OKAY" : "UNKNOWN")));
+  if (status > 0) {
+    buf_ptr += status;
+    buf_len -= status;
+  }
```

Comment:
You could assign severity to a separate string variable, use the append function also for this, and add `,\n` to the latter key values.

PS. I wonder though why both `\n` and `,` separators are needed, isn't one of them enough?